From 0aa11b5c8a91f4fb579eaf990c37fb3da0dc7368 Mon Sep 17 00:00:00 2001
From: Michael Smith <michael.smith@puppetlabs.com>
Date: Mon, 21 Dec 2015 13:58:33 -0800
Subject: [PATCH] CMake Bug 0015873 Workaround

https://cmake.org/Bug/view.php?id=15873

On SLES 12, `select` sometimes hangs. This patch disables using select
12 to avoid the issue, at the cost of slower CMake.
---
 Source/kwsys/ProcessUNIX.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/kwsys/ProcessUNIX.c b/Source/kwsys/ProcessUNIX.c
index 1be6d02..c5698d8 100644
--- a/Source/kwsys/ProcessUNIX.c
+++ b/Source/kwsys/ProcessUNIX.c
@@ -110,7 +110,7 @@ static inline void kwsysProcess_usleep(unsigned int msec)
  * without select().
  */
 #if !defined(__BEOS__) && !defined(__VMS) && !defined(__MINT__)
-# define KWSYSPE_USE_SELECT 1
+# define KWSYSPE_USE_SELECT 0
 #endif
 
 /* Some platforms do not have siginfo on their signal handlers.  */
-- 
2.6.4

